### PR TITLE
feat: precompiled binaries

### PR DESCRIPTION
**Pull request by fahchen**
**NOTE:** ~~This PR is based on https://github.com/tessi/wasmex/pull/320, I'll rebase it after merged.~~

fix: https://github.com/tessi/wasmex/issues/318
Check the final result at [the release](https://github.com/fahchen/wasmex/releases/tag/v0.6.1) and [the workflow](https://github.com/fahchen/wasmex/actions/runs/2044013594).

~~The unresolved problem is the `wasmer` is not support `arm7hf` (See the issue at https://github.com/wasmerio/wasmer/issues/2324), **this breaks the checksum step on [the release flow](https://hexdocs.pm/rustler_precompiled/precompilation_guide.html#the-release-flow)**~~ ✅.
As https://github.com/philss/rustler_precompiled/pull/8 is merged and released, the above problem is resolved. 🎉

demo:
[asciicast](https://asciinema.org/a/480738)

### CODE REVIEW

**Comment by fahchen** [comment]
As https://github.com/philss/rustler_precompiled/pull/8 is merged and released, this PR is ready to go. 🎉🎉🎉 @tessi

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        nif:
+          - "2.14"
+          - "2.15"
+          - "2.16"
+        job:
+          # https://github.com/wasmerio/wasmer/issues/2324
+          # - { target: arm-unknown-linux-gnueabihf , os: ubuntu-20.04 , use-cross: true }
```
Comment (tessi) [inline .github/workflows/release.yaml:38]: :thanks for documenting this exception!

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+      with:
+        name: ${{ steps.rename.outputs.LIB_FINAL_NAME }}
+        path: ${{ steps.rename.outputs.LIB_FINAL_PATH }}
+
+    - name: Publish archives and packages
+      uses: softprops/action-gh-release@v1
+      with:
+        files: |
+          ${{ steps.rename.outputs.LIB_FINAL_PATH }}
+      if: startsWith(github.ref, 'refs/tags/')
```
Comment (tessi) [inline .github/workflows/release.yaml:162]: Ah, do I read this correctly? This creates a GH release automatically is a commit on `main` is tagged, right?

**Comment by tessi** [comment]
Very solid work @fahchen. love it! I've just a question left to make sure I understand the release script correctly

File: .github/workflows/release.yaml (hunk 2 above)
Comment (fahchen) [reply .github/workflows/release.yaml:162]: The rules are below:
1. Always run on main branch
2. Run if "native" path changed
3. Tags will always run


 It creates a GH release only when a new tag is pushed, it only runs building test on `main` branch pushes or native folder changes.

| branch | building test | creating a release |
|---|---|---|
| `master` | ✅ | ❌ |
| `feat-branch` | ❌ | ❌ |
| `feat-branch-with-native-changes` | ✅ | ❌ |
| `v1.0.1` | ✅ | ✅ |

Comment (fahchen) [reply .github/workflows/release.yaml:162]: > Ah, do I read this correctly? This creates a GH release automatically is a commit on `main` is tagged, right?

It only creates a GH release automatically when a tag pushed.

Comment (tessi) [reply .github/workflows/release.yaml:162]: Thanks for the explanation!

**Comment by tessi** [comment]
I will resolve merge conflicts and merge this later today. Will create a new release right after 🎉

**Comment by tessi** [comment]
Time to celebrate! Thanks @fahchen for your hard work

**Pull request merged**